client/packages/invoices/src/Returns/modals/InboundReturn/InboundReturn.tsx

```diff
@@ -84,22 +100,43 @@ export const InboundReturnEditModal = ({
     alertRef?.current?.scrollIntoView({ behavior: 'smooth' });
   };
 
+  const CancelButton = <DialogButton onClick={onClose} variant="cancel" />;
+  const BackButton = (
+    <DialogButton onClick={() => onChangeTab(Tabs.Quantity)} variant="back" />
+  );
+  const NextStepButton = (
+    <DialogButton
+      onClick={handleNextStep}
+      variant="next"
+      disabled={!lines.length}
+      customLabel={t('button.next-step')}
+    />
+  );
+  const OkButton = <DialogButton onClick={onOk} variant="ok" />;
+  const OkAndNextButton = (
+    <DialogButton
+      onClick={handleNextItem}
+      variant="next"
+      disabled={currentTab !== Tabs.Reason}
+    />
+  );
+
   return (
     <TableProvider createStore={createTableStore}>
       <Modal
         title={t('heading.return-items')}
-        cancelButton={<DialogButton onClick={onClose} variant="cancel" />}
+        cancelButton={currentTab === Tabs.Quantity ? CancelButton : BackButton}
         // zeroQuantityAlert === warning implies all lines are 0 and user has
         // been already warned, so we act immediately to update them
-        nextButton={
-          currentTab === Tabs.Quantity && zeroQuantityAlert !== 'warning' ? (
-            <DialogButton onClick={handleNext} variant={'next'} />
-          ) : undefined
-        }
         okButton={
-          currentTab === Tabs.Reason || zeroQuantityAlert === 'warning' ? (
-            <DialogButton onClick={onOk} variant="ok" />
-          ) : undefined
+          currentTab === Tabs.Quantity && zeroQuantityAlert !== 'warning'
+            ? NextStepButton
+            : OkButton
+        }
+        nextButton={
+          modalMode === ModalMode.Update && loadNextItem
+            ? OkAndNextButton
+            : undefined
```

Comment:
Changed the order here so match the order they get displayed by the `useDialog` modal: "Cancel" -> "Ok" -> "Next"